src/ds5/ds5-options.h

```diff
@@ -415,7 +411,22 @@ namespace librealsense
             }
             _hwm.send(cmd);
         };
-        virtual float query() const override { return _value; };
+        virtual float query() const override 
+        { 
+            auto offset = 0;
+            if (_option == RS2_OPTION_AUTO_GAIN_LIMIT_TOGGLE)
+                offset = 4;
+            command cmd(ds::AUTO_CALIB);
+            cmd.param1 = 5;
+            auto res = _hwm.send(cmd);
+            if (res.empty())
+                throw invalid_value_exception("auto_exposure_limit_option::query result is empty!");
+
+            auto limit_val = static_cast<float>(*(reinterpret_cast<uint32_t*>(res.data() + offset)));
+            if (limit_val == 0)
+                return 0;
+            return 1;
+        };
         virtual option_range get_range() const override { return _toggle_range; };
         virtual bool is_enabled() const override { return true; }
         virtual const char* get_description() const override { return _description; };
```

Comment:
Check if those already exist in the base class

Reply:
Those are not implemented in base class